### ISSUE TITLE
dotnet list package: fails when using <PROJECT | SOLUTION> argument

### ISSUE THREAD

**Issue by josundt** [NuGet/Home#13307]
### NuGet Product Used

dotnet.exe

### Product Version

dotnet 8.0.200 (NuGet 6.9.1.3)

### Worked before?

Don't know

### Impact

I'm unable to use this version

### Repro Steps & Context

With the `dotnet list` command; `<PROJECT | SOLUTION>` (project or solution file path) is an optional argument meant to be supported:

```bash
> dotnet list -h

Description:
  List references or packages of a .NET project.

Usage:
  dotnet list [<PROJECT | SOLUTION>] [command] [options]

Arguments:
  <PROJECT | SOLUTION>  The project or solution file to operate on. If a file is not specified, the command will search the current directory for one. [default: /home/me/repos/myrepo/src]

Options:
  -?, -h, --help  Show command line help.


Commands:
  package    List all package references of the project or solution.
  reference  List all project-to-project references of the project.
```

Unfortunately it does not work in latest .NET SDK (8.0.200).

## Scenarios ##
1. The `<PROJECT | SOLUTION>` argument is a **DIRECTORY** path
2. The `<PROJECT | SOLUTION>` argument is **NOT** provided
3. The `<PROJECT | SOLUTION>` argument is a **SOLUTION** file path
4. The `<PROJECT | SOLUTION>` argument is a **PROJECT** file path

## Expected behavior ##
For the 4 described scenarios, the following strategy is applied by most of the .NET SDK commands when determining which .NET project files to process:

1. Scenario 1
    * If **multiple** `SOLUTION` files are found in directory, message _"Specify which project or solution file to use..."_ + `error exit code`
    * Else if a **single** `SOLUTION` file is found in directory, use that solution file/all projects referenced in solution file.
    * Else if **multiple** `PROJECT` files are found in directory, message _"Specify which project or solution file to use..."_ + `error exit code`
    * Else if a **single** `PROJECT` file is found in directory, use that file.
2. Scenario 2: Same as 1. using `cwd` as directory
3. Scenario 3: Use the solution file/all projects referenced in solution file.
4. Scenario 4: Use the project file.

## Actual behavior ##
_NB! The actual behavior was tested on both Windows and Linux_

1. Scenario 1: **Actual behavior does NOT meet expected behavior when using relative paths**:
    ```bash
    > dotnet list src/ package
    error: Unexpected failure reading NuGet.Config. Path: '/home/me/repos/myrepo/src/MySolution.sln/src/nuget.config'.
    error:   Could not find a part of the path '/home/me/repos/myrepo/src/MySolution.sln/src'.
    
    
    Usage: NuGet.CommandLine.XPlat.dll package list [arguments] [options]
    
    Arguments:
      <PROJECT | SOLUTION>  A path to a project, solution file or directory.
    
    ...
    ```
    _PS! Notice the malformed expected nuget.config file and folder path_

2. Scenario 2: **Actual behavior consistent with expected.**

3. Scenario 3: **Actual behavior does NOT meet expected behavior when using relative path**:
    ```bash
    > dotnet list MySolution.sln package
    error: Unexpected failure reading NuGet.Config. Path: '/home/me/repos/myrepo/src/MySolution.sln/nuget.config'.
    error:   The file '/home/me/repos/myrepo/src/MySolution.sln' already exists.
    
    Usage: NuGet.CommandLine.XPlat.dll package list [arguments] [options]
    
    Arguments:
      <PROJECT | SOLUTION>  A path to a project, solution file or directory.
    
    ...
    ```
    _PS! Notice the malformed expected nuget.config path - and also the strange error that the file already exists_

4. Scenario 4: **Actual behavior consistent if the specified project file is in `cwd`, otherwise path needs to be absolute.** 
    Example when specifying a project file in a different folder than `cwd`:
    ```bash
    > dotnet list MyProject/MyProject.csproj package
    error: Unexpected failure reading NuGet.Config. Path: '/home/me/repos/myrepo/src/MyProject/MyProject.csproj/nuget.config'.
    error:   The file '/home/me/repos/myrepo/src/MyProject/MyProject.csproj' already exists.
    
    Usage: NuGet.CommandLine.XPlat.dll package list [arguments] [options]
    
    Arguments:
      <PROJECT | SOLUTION>  A path to a project, solution file or directory.
    
    ...
    ```
    _PS! Notice the malformed expected nuget.config path - and also the strange error that the file already exists_

## Expected NuGet.Config search strategy ##
The reason for the problems described above seem to be that the tool does not correctly determine where to find a NuGet.Config file. I therefore also list what I believe should be the expected NuGet config file search strategy for the 4 described scenarios:

1. Scenario 1: Search the directory and recursively search parent directories until NuGet config file is found.
2. Scenario 2: Search `cwd` and recursively search parent directories until NuGet config file is found.
3. Scenario 3: Search the directory containing the solution file and recursively search parent directories until NuGet config file is found.
4. Scenario 4: Search the directory containing the project file and recursively search parent directories until NuGet config file is found.

**Comment by martinrrm** [comment]
I'm able to reproduce this in 8.0.201 and 8.0.102
![image](https://github.com/NuGet/Home/assets/43253759/cdc7c7a7-f995-4a82-a06a-58412b3f75be)

![image](https://github.com/NuGet/Home/assets/43253759/dc44ac40-f73b-4cc2-b60a-8a06ebcc1ac1)

But it looks like it's fixed in 9.0.100

![image](https://github.com/NuGet/Home/assets/43253759/5b0b6ae6-5877-4592-9232-64e1fb36a725)


@nkolev92 Sorry but I don't see any recent change in this command, we even [closed one](https://github.com/NuGet/Home/issues/13236) that is close to this one, do you know if we fixed something like this for .NET 9?

@Nigusu-Allehu Is your work on search related to this area? Did you fix this?

**Comment by Nigusu-Allehu** [comment]
The fix to "`dotnet list not` recognizing relative paths" should be only available starting 8.0.3xx .(https://github.com/NuGet/Home/issues/12954#issuecomment-1946877023)

**Comment by nkolev92** [comment]
@martinrrm 
Does it repro on 8.0.3xx?

**Comment by martinrrm** [comment]
@nkolev92 It doesn't, looks like fix is on SDK side https://github.com/dotnet/sdk/pull/38306, can I close this issue?
![image](https://github.com/NuGet/Home/assets/43253759/b73d253f-a418-4abd-b6c7-20a7229c5ea9)

**Comment by nkolev92** [comment]
Yep, duplicate fixed in 8.0.3xx.

**Comment by martinrrm** [comment]
Closing, duplicate of https://github.com/dotnet/sdk/pull/38306
